scenarios/steps/presign.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+
+import os
+
+import sh
+import yaml
+import urllib.parse
+
+from behave import *
+from assertpy import assert_that
+
+from qingstor.sdk.config import Config
+from qingstor.sdk.service.qingstor import QingStor
+
+config = Config().load_user_config()
+qingstor = QingStor(config)
+test_config_file_path = os.path.abspath(
+    os.path.join(os.path.dirname(__file__), os.path.pardir)
+)
+with open(test_config_file_path + '/test_config.yaml') as f:
+    test_data = yaml.load(f)
+    f.close()
+bucket = qingstor.Bucket(test_data['bucket_name'], test_data['zone'])
+bucket.put()
+qsctl = sh.Command("qsctl")
+
+
+@given(u'a non-public bucket with objects')
+def step_impl(context):
+    context.input = context.table
+    bucket.put_object(context.input[0]["name"])
+    current_acl = bucket.get_acl()
+    is_public = 0
+    for v in current_acl["acl"]:
+        if v["grantee"]["name"] == "QS_ALL_USERS":
+            is_public = 1
+    assert_that(is_public == 0
+                ).is_equal_to(True)  # check whether the bucket is non-public
+
+
+@when(u'generate the link with signature')
+def step_impl(context):
+    context.output = qsctl(
+        "presign",
+        "qs://{bucket}/{filename}".format(
+            bucket=test_data["bucket_name"],
+            filename=context.input[0]["name"],
+        ),
+        "-e",
+    ).stdout.decode("utf-8")
+
+
+@then(u'the link should include parameters like signature, etc.')
+def step_impl(context):
+    result = urllib.parse.urlparse(context.output)
+    params = urllib.parse.parse_qs(result.query, True)
+    object_key = result.path
+    assert_that("signature" in params.keys() and
```

Comment:
`assert_that(params).contains("signature")` maybe better, you can read the assertpy's doc <https://github.com/ActivisionGameScience/assertpy> for more infomation.